crates/scroll/alloy/network/src/lib.rs

```diff
@@ -0,0 +1,228 @@
+
+    type Header = alloy_consensus::Header;
+
+    type TransactionRequest = scroll_alloy_rpc_types::ScrollTransactionRequest;
+
+    type TransactionResponse = scroll_alloy_rpc_types::Transaction;
+
+    type ReceiptResponse = scroll_alloy_rpc_types::ScrollTransactionReceipt;
+
+    type HeaderResponse = alloy_rpc_types_eth::Header;
+
+    type BlockResponse =
+        alloy_rpc_types_eth::Block<Self::TransactionResponse, Self::HeaderResponse>;
+}
+
+impl TransactionBuilder<Scroll> for ScrollTransactionRequest {
+    fn chain_id(&self) -> Option<ChainId> {
+        self.as_ref().chain_id()
+    }
+
+    fn set_chain_id(&mut self, chain_id: ChainId) {
+        self.as_mut().set_chain_id(chain_id);
+    }
+
+    fn nonce(&self) -> Option<u64> {
+        self.as_ref().nonce()
+    }
+
+    fn set_nonce(&mut self, nonce: u64) {
+        self.as_mut().set_nonce(nonce);
+    }
+
+    fn input(&self) -> Option<&Bytes> {
+        self.as_ref().input()
+    }
+
+    fn set_input<T: Into<Bytes>>(&mut self, input: T) {
+        self.as_mut().set_input(input);
+    }
+
+    fn from(&self) -> Option<Address> {
+        self.as_ref().from()
+    }
+
+    fn set_from(&mut self, from: Address) {
+        self.as_mut().set_from(from);
+    }
+
+    fn kind(&self) -> Option<TxKind> {
+        self.as_ref().kind()
+    }
+
+    fn clear_kind(&mut self) {
+        self.as_mut().clear_kind();
+    }
+
+    fn set_kind(&mut self, kind: TxKind) {
+        self.as_mut().set_kind(kind);
+    }
+
+    fn value(&self) -> Option<U256> {
+        self.as_ref().value()
+    }
+
+    fn set_value(&mut self, value: U256) {
+        self.as_mut().set_value(value);
+    }
+
+    fn gas_price(&self) -> Option<u128> {
+        self.as_ref().gas_price()
+    }
+
+    fn set_gas_price(&mut self, gas_price: u128) {
+        self.as_mut().set_gas_price(gas_price);
+    }
+
+    fn max_fee_per_gas(&self) -> Option<u128> {
+        self.as_ref().max_fee_per_gas()
+    }
+
+    fn set_max_fee_per_gas(&mut self, max_fee_per_gas: u128) {
+        self.as_mut().set_max_fee_per_gas(max_fee_per_gas);
+    }
+
+    fn max_priority_fee_per_gas(&self) -> Option<u128> {
+        self.as_ref().max_priority_fee_per_gas()
+    }
+
+    fn set_max_priority_fee_per_gas(&mut self, max_priority_fee_per_gas: u128) {
+        self.as_mut().set_max_priority_fee_per_gas(max_priority_fee_per_gas);
+    }
+
+    fn gas_limit(&self) -> Option<u64> {
+        self.as_ref().gas_limit()
+    }
+
+    fn set_gas_limit(&mut self, gas_limit: u64) {
+        self.as_mut().set_gas_limit(gas_limit);
+    }
+
+    fn access_list(&self) -> Option<&AccessList> {
+        self.as_ref().access_list()
+    }
+
+    fn set_access_list(&mut self, access_list: AccessList) {
+        self.as_mut().set_access_list(access_list);
+    }
+
+    fn complete_type(&self, ty: ScrollTxType) -> Result<(), Vec<&'static str>> {
+        match ty {
+            ScrollTxType::L1Message => Err(vec!["not implemented for deposit tx"]),
```

Comment:
Does this mean we can't use the builder pattern with the `ScrollTxType::L1Message` variant?

Reply:
yeah, 

for primitive Tx, it still use the primitive checker. for TxL1Message we don't implement it that this is following op 
```
    fn complete_type(&self, ty: TxType) -> Result<(), Vec<&'static str>> {
        match ty {
            TxType::Legacy => self.complete_legacy(),
            TxType::Eip2930 => self.complete_2930(),
            TxType::Eip1559 => self.complete_1559(),
            TxType::Eip4844 => self.complete_4844(),
            TxType::Eip7702 => self.complete_7702(),
        }
    }
  ```